1_fetch/src/fetch_sites.R

```diff
@@ -22,30 +22,32 @@ fetch_states_from_sf <- function(sf_ind) {
 fetch_sites_from_states <- function(ind_file, state_cds, dates, stream_params) {
 
   # Cast wide net for all NWIS sites with stage data that fall within that bbox
-  sites_df <- data.frame()
-  for(cd in state_cds) {
-    sites_df_cd <- dataRetrieval::whatNWISdata(stateCd = cd, parameterCd = stream_params$stage, service = "uv")
-    sites_df <- rbind(sites_df, sites_df_cd)
-  }
+  sites_df <- bind_rows(lapply(state_cds, function(cd) {
+    dataRetrieval::whatNWISdata(stateCd = cd, parameterCd = stream_params$stage, service = "uv") %>%
+      dplyr::select(site_no, station_nm, dec_lat_va, dec_long_va, site_tp_cd, end_date)
+  }))
 
   # Get NWS flood stage table
   nws_flood_stage_list <- jsonlite::fromJSON("https://waterwatch.usgs.gov/webservices/floodstage?format=json")
   nws_flood_stage_table <- nws_flood_stage_list[["sites"]]
 
   # Filtering applied to every storm
-  sites_df <- sites_df %>%
+  sites_filtered <- sites_df %>%
     # Filter out any sites that don't have flood stage data from NWS
-    left_join(nws_flood_stage_table) %>%
-    filter(!is.na(flood_stage)) %>%
+    inner_join(nws_flood_stage_table, by='site_no') %>%
```

Comment:
It's not super important here, because the next thing we do is remove any rows with `is.na(flood_stage)`, but I changed it as I was going because inner_join seemed more in keeping with the spirit of what we were doing: `inner_join` only keeps and joins those rows that exist in both tables (in this case based on 'site_no'), and we're only interested in those sites for which we can get stage data AND that have flood stage information, so inner_join seemed best to me. In contrast, if we didn't have that filter on is.na(flood_stage), a `left_join` would be keeping all sites with stage data regardless of whether they're recognized by the NWS flood stage list or not, which seems suboptimal to me because we really want that intersection of stage data and flood stage info. But again, in this particular case the results are not changed by this switch.